app/hooks/handleUpload.ts

```diff
@@ -0,0 +1,123 @@
+import { useCallback } from 'react'
+import { FileItem } from '../credentialForm/form/types/Types'
+import { GoogleDriveStorage } from '@cooperation/vc-storage'
+import { uploadFilesToWAS } from '../utils/wasUpload'
+import { PortfolioItem } from '../credentialForm/form/Steps/Step3_uploadEvidence'
+
+interface HandleUploadOptions {
+  selectedFiles: FileItem[]
+  setValue: (field: string, value: any, options?: any) => void
+  setSelectedFiles: React.Dispatch<React.SetStateAction<FileItem[]>>
+  watch: <T>(name: string) => T
+  appInstanceDid?: any
+  hasZcap?: boolean
+  storage?: GoogleDriveStorage
+  useWas?: boolean
+}
+
+export function useHandleUpload({
+  selectedFiles,
+  setValue,
+  setSelectedFiles,
+  watch,
+  appInstanceDid,
+  hasZcap,
+  storage,
+  useWas = true,
+}: HandleUploadOptions) {
+  return useCallback(async () => {
+    try {
+      if (selectedFiles.length === 0) return
+
+      const filesToUpload = selectedFiles.filter(
+        f => !f.uploaded && f.file && f.name
+      )
+      if (filesToUpload.length === 0) return
+
+      if (
+        useWas &&
+        appInstanceDid &&
+        hasZcap &&
+        (appInstanceDid.keyPair || appInstanceDid.privateKeyMultibase)
```

Comment:
I'm not sure this line is necessary -- if there's an appInstanceDid, there must always be keypair etc